k4FWCore/components/IIOSvc.h

```diff
@@ -48,10 +43,10 @@ class IIOSvc : virtual public IInterface {
                                                     next()                     = 0;
   virtual std::shared_ptr<std::vector<std::string>> getCollectionNames() const = 0;
 
-  virtual std::shared_ptr<podio::ROOTWriter> getWriter()                                         = 0;
-  virtual void                               deleteWriter()                                      = 0;
-  virtual void                               deleteReader()                                      = 0;
-  virtual bool                               checkIfWriteCollection(const std::string& collName) = 0;
+  virtual podio::Writer& getWriter()                                         = 0;
+  virtual void           deleteWriter()                                      = 0;
+  virtual void           deleteReader()                                      = 0;
```

Comment:
Should these be named differently now that they are no longer pointer backed? Something like `resetWriter|resetReader`?

Reply:
I think it doesn't matter but the reason why they are needed is because services in Gaudi are not deleted so making the connection would be easier and technically they are being deleted.